File: src/features/migrations/dto/abort-migration.dto.ts
import { Transform } from 'class-transformer';
import { IsOptional, IsString } from 'class-validator';

export class AbortMigrationDto {
  @IsString()
  @Transform(({ value }) => (typeof value === 'string' ? value.trim() : value))
  migrationId: string;

  @IsString()
  @Transform(({ value }) => (typeof value === 'string' ? value.trim() : value))
  orgId: string;

  @IsOptional()
  @IsString()
  @Transform(({ value }) => (typeof value === 'string' ? value.trim() : value))
  reason?: string;
}

File: spring-boot-api/src/main/java/com/mytegroup/api/dto/migrations/AbortMigrationDto.java
package com.mytegroup.api.dto.migrations;

import jakarta.validation.constraints.NotBlank;

public record AbortMigrationDto(
    @NotBlank(message = "Migration ID is required")
    String migrationId,
    @NotBlank(message = "Organization ID is required")
    String orgId,
    String reason
) {
    public AbortMigrationDto {
        if (migrationId != null) {
            migrationId = migrationId.trim();
        }
        if (orgId != null) {
            orgId = orgId.trim();
        }
        if (reason != null) {
            reason = reason.trim();
        }
    }
}

